### PR TITLE
Kill Nose when ^C in sniffer

### PR DESCRIPTION
**Pull request by proteusvacuum**
**Kill Nose when ^C in sniffer**

@millerdev @benrudolph cc: @dannyroberts

### CODE REVIEW

**Comment by benrudolph** [comment]
perfect thanks 👃 🔫

File: scent.py [hunk 1]
```diff
@@ -1,9 +1,59 @@
...
+def py_files(filename):
+    return filename.endswith('.py') and not os.path.basename(filename).startswith('.')
+
+
+# Here we instruct the 'javascript_tests' runnable to be kicked off
+# when a .js file is changed
+@select_runnable('javascript_tests')
+@file_validator
+def js_files(filename):
+     return filename.endswith('.js') and not os.path.basename(filename).startswith('.')
```
Comment (dimagimon) [inline scent.py]: E111 indentation is not a multiple of four

**Comment by proteusvacuum** [comment]
Also added the ability to run mocha tests along with regular python tests.
eg: `sniffer -x --js-app_manager -x corehq.apps.app_manager:SomeTest` will watch both the javascript and python files. If you save a js file, it will run the js tests, if you save a python file, it will run the python tests.

**Comment by millerdev** [comment]
👍

**Comment by proteusvacuum** [comment]
@dimagi/team-commcare-hq You can now run JS tests in sniffer. One tool to rule them all :ring:

**Comment by benrudolph** [comment]
🙌 🙌

**Pull request merged**